server/internal/calc/euclidean_amd64.s

```diff
@@ -0,0 +1,29 @@
+#include "textflag.h"
+
+// func __euclideanDistance64(a, b []float64) float64
+TEXT ·__euclideanDistance64(SB), NOSPLIT, $0-56
+  MOVQ a+8(FP), DX  // Load the length of a into the DX register.
+  MOVQ a+0(FP), AX  // Load the pointer of the first array.
+  MOVQ b+24(FP), BX // Load the pointer of the second array.
+
+  VXORPD Y0, Y0, Y0 // Clear the accumulator registers.
+
+  LOOP:
+    VMOVUPD (AX), Y1  // Load the current 4 float64s from a into the 256-bit register Y1.
+    VMOVUPD (BX), Y2  // Load the current 4 float64s from b into the 256-bit register Y2.
+    VSUBPD Y2, Y1, Y1 // Subtract the 4 from a from the 4 from b and store the result in Y1.
```

Comment:
Is this actually subtracting a from b? I think the first two registers are reversed.